Tests/kaas/k8s-version-policy/k8s_version_policy.py

```diff
@@ -381,6 +388,80 @@ async def collect_cve_versions(session: aiohttp.ClientSession) -> set:
     return cfvs
 
 
+async def run_trivy_scan(image: str) -> dict:
+    """
+    Run Trivy scan on the specified image and return the results as a dictionary.
+
+    Args:
+        image (str): The Docker image to scan.
+
+    Returns:
+        dict: Parsed JSON results from Trivy.
+    """
+    try:
+        # Run the Trivy scan command
+        result = await asyncio.create_subprocess_exec(
+            'trivy',
+            'image',
+            '--format', 'json',
+            '--no-progress',
+            image,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+
+        stdout, stderr = await result.communicate()
+
+        if result.returncode != 0:
+            logger.error("Trivy scan failed: %s", stderr.decode().strip())
+            return {}
+
+        # Parse the JSON output from Trivy
+        return json.loads(stdout.decode())
+
+    except Exception as e:
+        logger.error("Error running Trivy scan: %s", e)
+        return {}
+
+
+async def get_k8s_pod_images(kubeconfig, context=None) -> list[str]:
+    """Get the list of container images used by all the pods in the Kubernetes cluster."""
+    cluster_config = await kubernetes_asyncio.config.load_kube_config(kubeconfig, context)
+
+    async with kubernetes_asyncio.client.ApiClient() as api:
+        v1 = kubernetes_asyncio.client.CoreV1Api(api)
+        pods = await v1.list_pod_for_all_namespaces(watch=False)
+
+        images = set()
+        for pod in pods.items:
+            for container in pod.spec.containers:
+                images.add(container.image)
+
+            if pod.spec.init_containers:
+                for container in pod.spec.init_containers:
+                    images.add(container.image)
+
+        return list(images)
+
+
+async def scan_k8s_images(kubeconfig, context=None) -> None:
+    """Scan the images used in the Kubernetes cluster for vulnerabilities."""
+    images_to_scan = await get_k8s_pod_images(kubeconfig, context)
```

Comment:
For my taste, `images_to_scan` should be the argument here. It would be the caller's responsibility to compose the two functions (and the caller's freedom to to something else entirely, which can be useful for unit tests).

Reply:
@mbuechse why is that? Isn't that the point of scanning all the images used here? It is coming automatically for the images working on our cluster and if somebody wants to reuse the function in the unit tests, they can manually provide the version for that

Reply:
Please, I don't see the need for arguing here. It's a super small change I'm asking, and it would improve the code. I provided clear reasoning as to why. So why argue? Telling people to make a copy of your function is not very polite, is it?